prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -56,39 +98,424 @@ def _retrieve_redfish_sensor_data(self) -> List[Any]:
+        except redfish_utilities.systems.RedfishSystemNotFoundError:
+            logger.debug("No system instances found.")
+            return processor_count, processor_data
+
+        for system_id in system_ids:
+            processor_uris: List[str] = []
+            # eg: /redfish/v1/Systems/1/Processors
+            processors_root_uri = processors_root_uri_pattern.format(system_id)
+            processor_members: List[Dict] = self.redfish_obj.get(processors_root_uri).dict[
+                "Members"
+            ]
+            for member in processor_members:
+                processor_uris.append(member["@odata.id"])
+            processor_count[system_id] = len(processor_uris)
+            processor_data_in_curr_system: List[Dict] = []
+            for processor_uri in processor_uris:
+                # eg: /redfish/v1/Systems/1/Processors/{1,2}
+                processor = self.redfish_obj.get(processor_uri).dict
+                processor_data_in_curr_system.append(
+                    {
+                        "processor_id": processor["Id"],
+                        "model": processor["Model"] or "NA",
+                        "health": processor["Status"]["Health"] or "NA",
+                        "state": processor["Status"]["State"],
+                    }
+                )
+            processor_data[system_id] = processor_data_in_curr_system
+        logger.debug("Processor count: %s", processor_count)
+        logger.debug("Processor data: %s", processor_data)
+        return processor_count, processor_data
+
+    def get_storage_controller_data(self) -> Tuple[Dict[str, int], Dict[str, List]]:
+        """Return storage controller data and count.
+
+        Returns:
+            storage_controller_count: Dict of system ids mapped to the total number
+            of storage controllers on that system.
+            storage_controller_data: Dict of system ids mapped to list of storage
+            controllers on that system.
+
+        Example of storage_controller_count: {"sys1": 2}
+
+        Example of storage_controller_data:
+        {
+            "sys1": [
+                {
+                    "storage_id": "STOR1",
+                    "controller_id": "sc0",
+                    "health": "OK",
+                    "state": "Enabled",
+                },
+                {
+                    "storage_id": "STOR2",
+                    "controller_id": "sc1",
+                    "health": "OK",
+                    "state": "Enabled",
+                },
+            ],
+        }
 
-        Passes the ttl value to the cache decorator at runtime.
         """
+        storage_controller_data: Dict[str, List] = {}
+        # storage controllers on each system
+        storage_controller_count: Dict[str, int] = {}
+        storage_root_uri_pattern = "/redfish/v1/Systems/{}/Storage"
+        logger.info("Getting storage controller data...")
```

Comment:
**nitpick:** 

Debug?

Reply:
IMO, it  should be an INFO level log.

Looking at [this doc page on logging](https://docs.python.org/3/library/logging.html#logging.INFO), it mentions:
- INFO: Confirmation that things are working as expected.
- DEBUG: Detailed information, typically only of interest to a developer trying to diagnose a problem.

So in this case, we are confirming that the storage controller data is being obtained and working correctly. 

What are your thoughts?

Reply:
Because not all the log which show this `Getting ... data` has the same level. I would like to unified them.

Reply:
Currently, all the logs for `Getting <metric> data` are the same INFO level if you check.